src/PhpImap/DataPartInfo.php

```diff
@@ -114,6 +114,12 @@ protected function convertEncodingAfterFetch(): string
             $this->data = $this->mail->decodeMimeStr(
                 (string) $this->data // Data to convert
             );
+
+            $this->data = $this->mail->convertToUtf8(
+                $this->data,
+                $this->charset
+            );
+            $this->charset = 'utf-8';
```

Comment:
Not sure, if we really need this line and if it makes sense to set it hard-coded here to UTF-8. Needs to be double-checked.

Reply:
Looks like as this isn't required. It also doesn't make any sense to always set it to UTF-8 AFTER converting the data as it maybe were before something else.